pyuvdata/uvflag/uvflag.py

```diff
@@ -923,6 +923,24 @@ def _set_type_waterfall(self):
         if not self.future_array_shapes:
             self._freq_array.form = ("Nfreqs",)
 
+    def check_lsts_against_times(self):
+        """Check that LSTs consistent with the time_array and telescope location."""
+        lsts = uvutils.get_lst_for_time(
+            self.time_array, *self.telescope_location_lat_lon_alt_degrees
+        )
+
+        if not np.allclose(
+            self.lst_array,
+            lsts,
+            rtol=self._lst_array.tols[0],
+            atol=self._lst_array.tols[1],
+        ):
+            warnings.warn(
+                "The lst_array is not self-consistent with the time_array and "
+                "telescope location. Consider recomputing with the "
+                "`set_lsts_from_time_array` method."
+            )
+
```

Comment:
Since you're using the same function in three places, would it make sense to have this as a standalone function that is called in each class?

Reply:
Maybe. I thought about that, but it's basically already just a call to the utility method, so the only thing that'd save me from copying is the warning, which seemed like a very small savings.

Reply:
Agreed, and I did think of that. I'm happy either way, but I guess the tipping point for me bringing it up was that it appeared not twice but three times, and the warning message should be consistent between them all. If we end up changing it for some reason in one place, we have the cognitive overhead of having to change it everywhere.

Reply:
I made the change, I think you were right, it simplified some things. It also turned out there were two other similar checks in `uvh5` and `uvfits` that I could also replace with the utility method. I also realized that the standard radian tolerance was being redefined in several separate places and that the one in UVFlag had not been updated when we reduced that tolerance. So I moved it to be defined in the utility module as well.

Working through this also made me aware that the astrometry library is not being plumbed through all the places `get_lst_for_time` is called. I'll fix that in a future PR.